USBPcapDriver/USBPcapURB.c

```diff
@@ -673,41 +673,52 @@ VOID USBPcapAnalyzeURB(PIRP pIrp, PURB pUrb, BOOLEAN post,
             }
             packetHeader->header.transfer = USBPCAP_TRANSFER_ISOCHRONOUS;
 
+            transferLength = 0;
             if (transfer->TransferFlags & USBD_TRANSFER_DIRECTION_IN)
             {
                 if (post == TRUE)
                 {
                     /* Read from device, return from controller */
-                    attachData = TRUE;
-                }
-                else
-                {
-                    /* Read from device, on its way to controller */
-                    attachData = FALSE;
+                    if (transfer->TransferBuffer)
+                    {
+                        /* return packet buffers only if data has been transferred */
```

Comment:
Can you comment here that the TransferBufferLength, when the transfer returns from controller is not the actual buffer length but rather a sum of bytes that were read from the device?